### PR TITLE
Piggyback darwin attribute cache subscription

### PR DESCRIPTION
**Pull request by kpark-apple**
#### Problem
* Current Darwin attribute cache requires a dedicated subscription, and it is not possible to share a single wildcard subscription for both attribute cache and regular subscription.

#### Change overview
* Piggyback CHIPAttributeCacheContainer on CHIPDevice subscription
  instead of using a dedicated subscription.
* CHIPAttributeCacheContainer API for subscription was removed.
* CHIPDevice subscription API added optional CHIPAttributeCacheContainer object parameter.

#### Testing
* All tests were updated to use the new API.

### CODE REVIEW

File: src/darwin/Framework/CHIP/CHIPDevice.mm [hunk 1]
```diff
@@ -323,9 +337,25 @@ - (void)subscribeWithQueue:(dispatch_queue_t)queue
...
         = (params != nil) && (params.keepPreviousSubscriptions != nil) && [params.keepPreviousSubscriptions boolValue];
 
-    auto callback = std::make_unique<SubscriptionCallback>(queue, reportHandler, subscriptionEstablishedHandler);
-    auto readClient = std::make_unique<ReadClient>(InteractionModelEngine::GetInstance(), device->GetExchangeManager(),
-        callback->GetBufferedCallback(), ReadClient::InteractionType::Subscribe);
+    std::unique_ptr<SubscriptionCallback> callback;
+    std::unique_ptr<ReadClient> readClient;
+    std::unique_ptr<AttributeCache> attributeCache;
+    if (attributeCacheContainer) {
+        __weak CHIPAttributeCacheContainer * weakPtr = attributeCacheContainer;
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/CHIPDevice.mm:344]: So in terms of ownership, once no one is holding on to the `CHIPAttributeCacheContainer` the subscription will be killed off, right?  Or am I misunderstanding how this all works?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/CHIPDevice.mm (hunk 1 above)
Comment (kpark-apple) [reply src/darwin/Framework/CHIP/CHIPDevice.mm:344]: Good point. Your understanding is correct. I applied the same logic as was used with previous `CHIPAttributeCacheContainer` API: if the cache is no longer used, the subscription stops. Now with this API, there is a separate report handler associated with it, I wonder if we should continue the subscription even if the cache is not used any more. What do you think?

Comment (kpark-apple) [reply src/darwin/Framework/CHIP/CHIPDevice.mm:344]: Per discussion, I'll remove this behavior and keep the subscription even if `CHIPAttributeCacheContainer` object is released.

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform change.

**Pull request merged**